mood-analysis.rb

```diff
@@ -19,6 +23,60 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def happy_days(words)
+  entries = 0
+  happy = 0
+  strip = []
+
+  words.each do |word|
+    strip << strip_punctuation(word)
+  end
+
+  strip.each do |string|
+    if happy < 3
+      entries += 1
+      if analyze_mood(string) == ":-)"
+        happy += 1
+      end
+    elsif happy == 3
+      puts "It takes #{entries} entries for 3 happy days to occur."
+    else
+      puts "There were less than 3 happy days in #{entries} entries. :-("
+    end
+  end
+end
```

Comment:
This method would be better if it just returned a fixnum, not a string and then you can print the result of the method call in a string. This would isolate the return value and allow us to do other things with this information instead of being constrained to just saying how many days it takes for three happy days.